### PR TITLE
issue/1594: updated inview to version 2017-07-11

### PR DESCRIPTION
**Pull request by oliverfoster**
#1594 update to latest version
#1591 fix android setTimeout throttling
#1452 fix bookmarking negative area issue

* fixes android setTimeout throttling
* stop inview events propagating
* use window.innerHeight and width to calculate window size
* fixes negative area issue #1452

### CODE REVIEW

File: src/core/js/libraries/inview.js [hunk 1]
```diff
@@ -392,6 +420,7 @@
             var body = $("body")[0].getBoundingClientRect();
             //make sure to get height and width independently if getBoundingClientRect doesn't return height and width;
             measurements.supplimentDimensions = (body.width === undefined);
+            loop.hasRaf = (window.requestAnimationFrame && window.cancelAnimationFrame);
```
Comment (NayanKhedkar) [inline src/core/js/libraries/inview.js]: is it compatible with all browsers ex. Mozilla has window.mozRequestAnimationFrame.

Comment (oliverfoster) [inline src/core/js/libraries/inview.js]: :/ are you using firefox 10?

Comment (oliverfoster) [inline src/core/js/libraries/inview.js]: either way, this code is to check if raf is available and if not use ```setTimeout``` and ```clearTimeout``` instead.

Comment (moloko) [inline src/core/js/libraries/inview.js]: Firefox ESR is the lowest version we support - as of this month that is v52.x

Comment (NayanKhedkar) [inline src/core/js/libraries/inview.js]: 😃  i am not using firefox 10 , i was giving example of it 
I mean to say can we write this fallback or doesn't need to write this 
window.requestAnimationFrame = window.requestAnimationFrame
    || window.mozRequestAnimationFrame
    || window.webkitRequestAnimationFrame
    || window.msRequestAnimationFrame

Comment (oliverfoster) [inline src/core/js/libraries/inview.js]: 👍 yup, definitely no longer required

**Comment by moloko** [comment]
I'll kick the tyres on this a bit later today

**Pull request merged**